lib/exhal/assertions.ex

```diff
@@ -67,6 +67,13 @@ defmodule ExHal.Assertions do
   end
 
   @doc"""
+    Checks if a link is present in a rendered document.
+  """
+  def has_link?(json_map, rel) do
+    Kernel.get_in(json_map, ["_links", rel])
```

Comment:
Our implementation does not remove empty links from the `Document`, it removes them when the document is rendered (with `to_json_map`), so that test won't work. 

Removing empty links from the `Document` entirely would be an different implementation. I'll see how difficult that might be.